### ISSUE TITLE
Factoring out initqueue into its own dracut module for systemd-enabled initrds

### ISSUE THREAD

**Issue by LaszloGombos** [dracut-ng/dracut-ng#187]
An idea has been floating around to support a dracut configuration where systemd dracut modules are included with the exception of the dracut-systemd module. This would remove the initqueue from initrd and just use systemd to schedule.

Generating such as initrd is already supported by adding `-o dracut-systemd` to the dracut command line option, but such an initrd might not be fully functional. Systemd itself however supports this use case as some initrd generators (such as mkosi, mkinitramfs) generate initrds without an extra initqueue.

Mainly dropping issue this here just for visibility and to at least not work against this potential project goal. I myself not planning to work on it, but perhaps someone else is interested.

CC @lnykryn @pvalena

**Comment by dtardon** [comment]
I don't think that disablement of the whole `dracut-systemd` module is desirable. Most of the module's content is useful--even needed--stuff:
* handling of `rd.break=X` (`dracut-cmdline.service`, `dracut-pre-udev.service`, etc.)
* unpacking of "exitrd" on system shutdown (`dracut-shutdown.service`)

It would make a lot more sense to move the initqueue to a separate module first and _only then_ think about making that module optional. (I completely agree with the sentiment, though. Having two mostly independent and overlapping scheduling mechanisms increases complexity.)
